src/BasicStepperDriver.cpp

```diff
@@ -264,6 +273,12 @@ void BasicStepperDriver::calcStepPulse(void){
     }
     steps_remaining--;
     step_count++;
+    
+    // absolute position
+    if (dir_state)
+        step_pos_abs++;
+    else
+        step_pos_abs--;
```

Comment:
Let's find a way to avoid having to read/write these two variables here, because they are not needed for the calculation this method has to do, and `calcStepPulse()` execution time limits the maximum step frequency.

I think this could be done on-demand in the get/set methods using the current state: at any given moment, the absolute position is `(absolute position at start) +- (step_count)`.